src/client/src/consumer.rs

```diff
@@ -579,22 +580,19 @@ impl Record {
         self.offset
     }
 
-    pub fn to_string_lossy(&self) -> String {
-        String::from_utf8_lossy(self.as_ref()).to_string()
+    /// Returns the contents of this Record's key, if it exists
+    pub fn key(&self) -> Option<&[u8]> {
+        self.record.key.as_ref().map(|it| it.as_ref())
```

Comment:
Actually, key and value should move to `DefaultRecord`.  
Can use `Deref` to delegate methods

Reply:
Hm, I think using Deref for anything except smart-pointers is [considered an anti-pattern](https://github.com/rust-unofficial/patterns/blob/master/anti_patterns/deref.md), but I can make this method just forward to the `key()` method of `DefaultRecord` 👍

Reply:
Regarding `Deref`.  I disagree.  It's being used frequently and I see no problem here.
Alternative, then just implement `inner()` which will return DefaultRecord.  Caller just need to do extra hop

Creating delegation method is really anti-pattern in my opinion.